docs/index.asciidoc

```diff
@@ -0,0 +1,219 @@
+
+`%{?foo}` is a named skip field.
+
+*Append field notation:* +
+The value is appended to another value or stored if its the first field seen. +
+The key is prefixed with a `+`. +
+The final value is stored in the Event using the key. +
+
+[NOTE]
+====
+The delimiter found before the field is appended with the value. +
+If no delimiter is found before the field, a single space character is used.
+====
+
+`%{+some_field}` is an append field. +
+`%{+some_field/2}` is an append field with an order modifier.
+
+An order modifier, `/digits`, allows one to reorder the append sequence. +
+e.g. for a text of `1 2 3 go`, this `%{+a/2} %{+a/1} %{+a/4} %{+a/3}` will build a key/value of `a => 2 1 go 3` +
+Append fields without an order modifier will append in declared order. +
+e.g. for a text of `1 2 3 go`, this `%{a} %{b} %{+a}` will build two key/values of `a => 1 3 go, b => 2` +
+
+*Indirect field notation:* +
+The found value is added to the Event using the found value of another field as the key. +
+The key is prefixed with a `&`. +
+`%{&some_field}` - an indirect field where the key is indirectly sourced from the value of `some_field`. +
+e.g. for a text of `error: some_error, some_description`, this `error: %{?err}, %{&err}` will build a key/value of `some_error => some_description`.
+
+[NOTE]
+for append and indirect field the key can refer to a field that already exists in the event before dissection.
+
+[NOTE]
+use a Skip field if you do not want the indirection key/value stored.
+
+e.g. for a text of `google: 77.98`, this `%{?a}: %{&a}` will build a key/value of `google => 77.98`.
+
+[NOTE]
+===============================
+append and indirect cannot be combined and will fail validation. +
+`%{+&something}` - will add a value to the `&something` key, probably not the intended outcome. +
+`%{&+something}` will add a value to the `+something` key, again probably unintended. +
+===============================
+
+*Delimiter repetition:* +
+In the source text if a field has variable width padded with delimiters, the padding will be ignored. +
+e.g. for texts of:
+....
+00000043 ViewReceiver  I
+000000b3 Peer          I
+....
+with a dissection of `%{a} %{b} %{c}`; the padding is ignored, `event.get([c]) -> "I"`
+
+[NOTE]
+====
+You probably want to use this filter inside an `if` block. +
+This ensures that the event contains a field value with a suitable structure for the dissection.
+====
+
+For example...
+....
+filter {
+  if [type] == "syslog" or "syslog" in [tags] {
+    dissect {
+      mapping => {
+        "message" => "%{ts} %{+ts} %{+ts} %{src} %{} %{prog}[%{pid}]: %{msg}"
+      }
+    }
+  }
+}
+....
+
+==== Configuration Options
+
+This plugin supports the following configuration options:
+
+* <<plugins-{type}s-{plugin}-options>>
+* <<plugins-{type}s-common-options>>
+
+[id="plugins-{type}s-{plugin}-options"]
+==== Dissect Filter Options
+
+The following configuration options are supported by the dissect filter:
```

Comment:
@ph I'm really sorry to bring this up now (should have noticed it earlier), but when I look at the long list of options for some plugins, I'm concerned that people might overlook the common options (yes, even though there is a link right above this...people don't read carefully). What if we were to remove line 160/161 and change line 150 to say:

`This plugin supports the following configuration options plus the <<plugins-{type}s-common-options>> described later.`